api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java

```diff
@@ -56,22 +57,39 @@
 	 * @throws DAOException exception thrown if error occurs while deleting the diagnosis
 	 */
 	void deleteDiagnosis(Diagnosis diagnosis) throws DAOException;
-	
+
 	/**
 	 * Gets all diagnoses for a given encounter
 	 *
-	 * @param encounter the specific encounter to get the diagnoses for.
-	 * @return list of diagnoses for an encounter
+	 * @deprecated since 2.5.0, use {@link #getDiagnosesForEncounter}
+	 *
+	 * @param encounter the encounter for which to fetch diagnoses
+	 * @return the list of (primary, confirmed) diagnoses for the given encounter
 	 */
+	@Deprecated
 	List<Diagnosis> getDiagnoses(Encounter encounter);
+	
+	/**
+	 * Gets diagnoses for an Encounter. When specified, this method only returns
+	 * primary or confirmed diagnoses.
+	 *
+	 * @param encounter the encounter for which to fetch diagnoses
+	 * @param primaryOnly whether to return only primary diagnoses
```

Comment:
Instead of duplicating this description, shouldn't we just use? `@see org.openmrs.api.DiagnosisService#getDiagnosesByEncounter(org.openmrs.Encounter, boolean, boolean)`